core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -65,10 +66,73 @@ int HIP::impl_is_initialized() {
 }
 
 void HIP::impl_initialize(InitializationSettings const& settings) {
+  const int hip_device_id = ::Kokkos::Impl::get_gpu(settings);
+
+  const auto& dev_info = Impl::HIPInternalDevices::singleton();
```

Comment:
If we are refactoring we should just get rid of that `Impl::HIPInternalDevices` altogether